.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,23 @@
+name: Deploy gh-pages
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 16
+    - run: npm install
+    - run: npm run build
+    - uses: JamesIves/github-pages-deploy-action@v4.3.3
+      with:
+        branch: gh-pages
+        folder: build
+        clean: true
```

Comment:
does this mean the history of gh-pages is reset?

Reply:
No, just that it deletes any other files on the branch before committing.